### PR TITLE
FEAT: Add system to move entities

### PR DESCRIPTION
**Pull request by AxelF44**
I've added the communication between all entities and network to move on all side the player !

### CODE REVIEW

File: R-TypeGame/SystemNetworkUpdate/UpdatePositionSystem/updatePositionSystem.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                            packet.getPayload()[4 + componentTypeLength];
+
+        float x;
+        std::memcpy(&x, &packet.getPayload()[5 + componentTypeLength], sizeof(float));
+
+        float y;
+        std::memcpy(&y, &packet.getPayload()[9 + componentTypeLength], sizeof(float));
+
+        ECS::SparseArray<IComponent> PositionComponentArray = reg.get_components<IComponent>(componentType);
+        Position2DComponent* position = dynamic_cast<Position2DComponent*>(PositionComponentArray[idxPacketEntities].get());
```
Comment (mathieurobert1) [inline R-TypeGame/SystemNetworkUpdate/UpdatePositionSystem/updatePositionSystem.cpp:36]: Check the size of PositionComponentArray to prevent segfault

**Review by mathieurobert1 — CHANGES_REQUESTED** [review]
Just a request to prevent segfault

**Review by mathieurobert1 — APPROVED** [review]

File: Engine/Client/Src/Network/NetworkClient.cpp [hunk 2]
```diff
@@ -28,7 +28,9 @@ void Network::Client::connect(MessageHandler callback, ECS::Registry& reg)
 
         asio::error_code ec;
         asio::read(*_tcp_socket, asio::buffer(&_token, sizeof(_token)), ec);
+        asio::read(*_tcp_socket, asio::buffer(&_idxPlayerComponent, sizeof(_idxPlayerComponent)), ec);
         std::cout << "Connected to server with token: " << "0x" << std::hex << std::setw(8) << std::setfill('0') << _token << std::dec << std::endl;
+        std::cout << "Connected to server with index player component: " << _idxPlayerComponent << std::endl;
```
Comment (Thomaltarix) [inline Engine/Client/Src/Network/NetworkClient.cpp:33]: Is this debug too ?

[the remark this answers is not in the document]
Comment (Thomaltarix) [reply Engine/Client/Src/Network/NetworkClient.cpp:33]: Pretty weird to display a connection token even in the terminal when it isn't debug

**Review by Thomaltarix — APPROVED** [review]
Just a little question, If it isn't debug, merge it, if not, modify it please

**Pull request merged**